File: python-src/get_cb_orbs_balance.py
from dotenv import load_dotenv
import os

from prime_sdk.client_services import PrimeServicesClient
from prime_sdk.services.portfolios import ListPortfoliosRequest
from prime_sdk.services.balances import ListPortfolioBalancesRequest

load_dotenv()

"""
Script to get the balance of ORBS tokens in Orbs LTD Coinbase Prime portfolio.
"""

def main():
    # Build the unified services client from env (uses PRIME_CREDENTIALS)
    client = PrimeServicesClient.from_env("PRIME_CREDENTIALS")

    # 1) Get a portfolio id (use PRIME_PORTFOLIO_ID if you already exported it)
    portfolio_id = os.getenv("PRIME_PORTFOLIO_ID")

    # 2) Fetch balances for that portfolio
    req = ListPortfolioBalancesRequest(portfolio_id=portfolio_id)
    resp = client.balances.list_portfolio_balances(req)

    for b in resp.balances:
        if b.symbol == "orbs":
            return float(b.amount)

File: ts-src/src/get_cb_orbs_balance.ts
import 'dotenv/config';
import { CoinbasePrimeClientWithServices } from '@coinbase-sample/prime-sdk-ts';

async function getOrbsBalance(): Promise<number | undefined> {
  // Load the portfolio ID from environment
  const portId = process.env.PRIME_PORTFOLIO_ID;
  if (!portId) {
    throw new Error("Missing environment variable: PRIME_PORTFOLIO_ID");
  }

  // Initialize the Coinbase Prime client using credentials from the environment
  const client = CoinbasePrimeClientWithServices.fromEnv("PRIME_CREDENTIALS");

  // Fetch portfolio balances
  const resp = await client.balances.listPortfolioBalances({ portfolioId: portId });

  // Find ORBS balance
  for (const b of resp.balances) {
    if (b.symbol?.toLowerCase() === "orbs") {
      return parseFloat(b.amount);
    }
  }

  // Return undefined if ORBS not found
  return undefined;
}

// Example usage
(async () => {
  try {
    const balance = await getOrbsBalance();
    console.log("ORBS balance:", balance);
  } catch (err) {
    console.error("Error fetching balance:", err);
  }
})();
